docs/core/extensions/httpclient-factory.md

```diff
@@ -127,6 +128,48 @@ The typed client is registered as transient with DI. In the preceding code, `Add
 > [!NOTE]
 > When registering a typed client with the `AddHttpClient<TClient>` method, the `TClient` type must have a constructor that accepts an `HttpClient` parameter. Additionally, the `TClient` type shouldn't be registered with the DI container separately.
 
+### Named and typed clients
+
+Named clients and typed clients have their own strengths and weaknesses. There is a way to combine these two client types to get the best both worlds.
+
+The primary use case is the following: Use the same typed client but against different domains. For example you have a primary and a secondary service and they expose the exact same functionality. That means you can use the same typed client to wrap the `HttpClient` usage to issue requests, process responses, and handle errors. The exact same code will be used but with different configurations (different base address, timeout, credentials, etc.).
+
+The following example uses the same `TodoService` typed client which was shown under the [typed clients](#typed-clients) section.
+
+First register the named and typed clients.
+
+:::code source="snippets/http/namedandtyped/Program.cs" range="1-25" highlight="9,14-15,19,23-24":::
+
+In the preceding code:
+
+- The first `AddHttpClient` call registers a `TodoService` typed client under the `primary` name. The underlying `HttpClient` points to the primary service and has a short timeout.
+- The second `AddHttpClient` call registers a `TodoService` typed client under the `secondary` name. The underlying `HttpClient` points to the secondary service and has a longer timeout.
+
+:::code source="snippets/http/namedandtyped/Program.cs" range="27-39" highlight="34-35,38-39":::
```

Comment:
Highlights start from the beginning of the rendered `range`, and are 1-based I believe. I'm not sure what you're trying to highlight:

![image](https://github.com/dotnet/docs/assets/7679720/23e137b2-c07f-49ac-961c-cdd91774c485)

Reply:
Just fixed them :D